<launch>

<!-- ARGUMENTS -->
  <!--Dimensions-->
  <arg name="leg_length"      value="1.0" />
  <arg name="frame_length_x"  value="1.5" />
  <arg name="frame_length_y"  value="1.0" />
  <arg name="frame_thickness" value="0.09" />

<!--Limits are set on joint_parameters.yaml-->

<!-- SAVE ARGUMENTS AS PARAMETERS. Only nessesary for some of them -->
<param name="/dimensions/leg_length"        value="$(arg leg_length)"/>
<param name="/dimensions/frame_length_x"    value="$(arg frame_length_x)"/>
<param name="/dimensions/frame_length_y"    value="$(arg frame_length_y)"/>
<param name="/dimensions/frame_thickness"   value="$(arg frame_thickness)"/>

<!--Loads parameters stored in YAML into the parameter server-->
<rosparam command="load" file="$(find robostilt_description)/config/parameters.yaml" />

<!--Loads URDF model into parameter server using arguments above-->
 <param name="robot_description"
  command="xacro --inorder '$(find robostilt_description)/urdf/robostilt.urdf.xacro' 
  leg_length:=$(arg leg_length)
  frame_length_x:=$(arg frame_length_x)
  frame_length_y:=$(arg frame_length_y)
  frame_thickness:=$(arg frame_thickness)
  "/>


</launch>

